e_os.h

```diff
@@ -544,6 +544,8 @@ struct servent *getservbyname(const char *name, const char *proto);
 # define CRYPTO_memcmp memcmp
 #endif
 
+#define IF_FAIL_ASSERT(x)     assert(x); if (!(x))
```

Comment:
Yes, I thought about the side effect issue and meant to put a comment to that effect in here.

Another approach I thought of was something like:

````
#define IF_FAIL_ASSERT(x, body)    \
    {\
        int assrt;\
        assert(assrt = (int)(x));\
       if (!assrt) \
           body\
    }
````
That solves the side-effect problem but I think makes the usage a bit messy:
````
    IF_FAIL_ASSERT(x == y, {
        SSLerr(MY_FUNC, ERR_R_INTERNAL_ERROR);
        goto err;
    })
````
Or another option:
````
#define IF_FAIL_ASSERT(x, body)    \
    {\
        int assrt;\
        assert(assrt = (int)(x));\
       if (!assrt) {

#define END_FAIL_ASSERT }}
````
Which would then look like this in usage:

````
    IF_FAIL_ASSERT(x == y)
        SSLerr(MY_FUNC, ERR_R_INTERNAL_ERROR);
        goto err;
    END_FAIL_ASSERT
````
I didn't particularly like that either so, on balance I thought it was better to just make sure it never has a side-effect (which I think is the case for all the libssl ones I did) - but I'm open to other suggestions.